nomad/structs/structs.go

```diff
@@ -1053,6 +1053,93 @@ func (t *Task) GoString() string {
 	return fmt.Sprintf("*%#v", *t)
 }
 
+// Set of possible states for a task.
+const (
+	TaskStatePending = "pending" // The task is waiting to be run.
+	TaskStateRunning = "running" // The task is currently running.
+	TaskStateDead    = "dead"    // Terminal state of task.
```

Comment:
Ah there is a confusion. The State can only be `pending`, `running`, `dead`. However there are a lot of ways a task can enter these states. The `TaskEvent` are those. If you think of it as a finite state machine, the State are the nodes and the Events are the transitions.